### PR TITLE
Adds the wizard's lightning bolt spell, and the magic mirror

### LINKED ISSUE #4613: Change Paper Text box to TMP
Paper currently uses unity Text not TMP Text so has limited tag options.

https://docs.unity3d.com/Packages/com.unity.ugui@1.0/manual/StyledText.html

compared to:

http://digitalnativestudios.com/textmeshpro/docs/rich-text/

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Adds the electrical arc effect. Uses [LightningBolt](https://assetstore.unity.com/packages/tools/particles-effects/lightning-bolt-effect-for-unity-59471) asset.
- Adds the lightning bolt spell. Continues #5160.
lightning bolt
- The wizard now spawns with a random name.
- Adds the magic mirror. Enables the wizard to change their name.
    - Includes wizard scene changes to reflect the addition.
- Paper can now fit in the ID slot.
- Wizards now spawn with an identity certificate instead of a station ID card.
- The Wizard 101 paper that spawns in the pocket is now populated.
- Paper item now uses TextMeshPro instead of default Unity Text. Fixes #4613

CL:Added lightning bolt spell and the magic mirror.

### CODE REVIEW

**Comment by corp-0** [comment]
Codacy is complaining but it has a point. Unity uses IEnumerators in a hacky way to make code run "asynchronically". In this case we can ignore the complain.

**Comment by Ruzihm** [comment]
@corp-0  Nothing is calling `StartCoroutine` on `MobSprite.StopElectrocutedOverlayAfter`, and nothing is yield returning with it so codacy is right. it doesn't need to return `IEnumerator` because is is never used as a coroutine.

Contrast with `PlayerSprites.StopElectrocutedOverlayAfter`, which does get called as a coroutine.

I'm guessing it is meant to be called with `StartCoroutine` in this case.

**Comment by NoooneyDude** [comment]
Ah. Good catch! I'll commit a fix.

File: UnityProject/Assets/Scripts/NPC/MobSprite.cs [hunk 1]
```diff
@@ -172,12 +173,50 @@ public void SetBurningOverlay()
...
+		public void EnableElectrocutedOverlay(float time = -1)
+		{
+			if (electrocutedOverlay == null)
+			{
+				AddOverlayGameObjects();
+			}
+
+			if (time > 0)
+			{
+				StopElectrocutedOverlayAfter(time);
```
Comment (Ruzihm) [inline UnityProject/Assets/Scripts/NPC/MobSprite.cs]:
```suggestion
				StartCoroutine(StopElectrocutedOverlayAfter(time));
```

**Review by Ruzihm — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/NPC/MobSprite.cs (hunk 1 above)
Comment (Ruzihm) [reply UnityProject/Assets/Scripts/NPC/MobSprite.cs]: Ah, so that's how to suggest changes. neat!

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/NPC/MobSprite.cs]: Cheers!

**Comment by corp-0** [comment]
my bad. I thought it was complaining because the return value was never used, but it was actually the whole thing never being called. Hooray for codacy

File: UnityProject/Assets/Scripts/NPC/MobSprite.cs [hunk 2]
```diff
@@ -172,12 +173,50 @@ public void SetBurningOverlay()
...
+		/// Disables the electrocuted overlay for the mob.
+		/// </summary>
+		public void DisableElectrocutedOverlay()
+		{
+			if (electrocutedOverlay == null)
+			{
+				return;
+			}
+
+			burningOverlay.StopOverlay();
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/MobSprite.cs]:
```suggestion
			electrocutedOverlay.StopOverlay();
```

**Pull request merged**